bot/cogs/source.py

```diff
@@ -96,8 +93,15 @@ def get_source_link(self, source_item: SourceType) -> Tuple[str, str, Optional[i
 
         return url, file_location, first_line_no or None
 
-    async def build_embed(self, link: str, source_object: SourceType, loc: str, first_line: Optional[int]) -> Embed:
+    async def build_embed(self, source_object: SourceType, ctx: commands.Context) -> Optional[Embed]:
         """Build embed based on source object."""
+        url, location, first_line = self.get_source_link(source_object)
+
+        # There is no URL only when bot can't fetch Tags cog
+        if not url:
+            await ctx.send("Unable to get `Tags` cog.")
+            return
```

Comment:
I'm not sure on this idea, but you could let `get_source_link` raise an exception when the cog isn't loaded (though perhaps something more specific than the `AttributeError` on `None` that it would normally raise) and then catch that in `source_command`. What do you think about this? Would that be clearer or just more confusing?

Reply:
I refactored this now. I found that converter already makes sure is cog there, so no check is required in function.